src/mongoc/mongoc-server-description-private.h

```diff
@@ -28,19 +28,9 @@
 /* represent a server or topology with no replica set config version */
 #define MONGOC_NO_SET_VERSION -1
 
-typedef enum
-   {
-      MONGOC_SERVER_UNKNOWN,
-      MONGOC_SERVER_STANDALONE,
-      MONGOC_SERVER_MONGOS,
-      MONGOC_SERVER_POSSIBLE_PRIMARY,
-      MONGOC_SERVER_RS_PRIMARY,
-      MONGOC_SERVER_RS_SECONDARY,
-      MONGOC_SERVER_RS_ARBITER,
-      MONGOC_SERVER_RS_OTHER,
-      MONGOC_SERVER_RS_GHOST,
-      MONGOC_SERVER_DESCRIPTION_TYPES,
-   } mongoc_server_description_type_t;
+/* this is used for declaring gSDAMTransitionTable and should be defined as the
+ * number of unique states in mongoc_server_description_type_t */
+#define MONGOC_SERVER_DESCRIPTION_TYPES 9
```

Comment:
`mongoc-topology-description.c`? That seems a bit odd given that it has a `MONGOC_SERVER_DESCRIPTION` prefix. If moved there, would you want it at the top of the C file below the `#include` lines?